engine/src/test/java/org/entando/entando/aps/system/services/cache/CacheInfoManagerTest.java

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright 2018-Present Entando S.r.l. (http://www.entando.com) All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+package org.entando.entando.aps.system.services.cache;
+
+import com.agiletec.aps.system.exception.ApsSystemException;
+import com.agiletec.aps.system.services.page.Page;
+import com.agiletec.aps.system.services.page.events.PageChangedEvent;
+import java.lang.annotation.Annotation;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import junit.framework.Assert;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.springframework.cache.Cache;
+import org.springframework.cache.CacheManager;
+
+/**
+ * @author E.Santoboni
+ */
+public class CacheInfoManagerTest {

```

Comment:
These tests look really good. Probably worth thinking about how we can generalize these tests to apply to all of the cache managers generically so we don't have to write the same test methods 50+ times.

Doesn't need to happen now but keep in mind.

Reply:
Clear concept.
This is the only case in which I thought to not insert a CacheWrapper, for the simple reason that this manager already manages cache.
We will keep this in mind for the other issues.